pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -0,0 +1,187 @@
+package image
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/diskfs/go-diskfs/filesystem"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/asset"
+)
+
+const (
+	initrdimg = "initrd"
+	rootfsimg = "rootfs"
+	vmlinuz   = "vmlinuz"
+	// pxeAssetsPath is the path where pxe files are created.
+	pxeAssetsPath = "pxe"
+)
+
+// AgentPXEFiles is an asset that generates the bootable image used to install clusters.
+type AgentPXEFiles struct {
+	imageReader isoeditor.ImageReader
+	cpuArch     string
+}
+
+var _ asset.WritableAsset = (*AgentPXEFiles)(nil)
+
+// Dependencies returns the assets on which the AgentPXEFiles asset depends.
+func (a *AgentPXEFiles) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&Ignition{},
+		&BaseIso{},
+	}
+}
+
+// Generate generates the image files for PXE asset.
+func (a *AgentPXEFiles) Generate(dependencies asset.Parents) error {
+	ignition := &Ignition{}
+	dependencies.Get(ignition)
+
+	baseImage := &BaseIso{}
+	dependencies.Get(baseImage)
+
+	isoGetter := newGetIso(GetIsoPluggable)
+	isoPath, err := isoGetter.getter()
+	if err != nil {
+		return err
+	}
```

Comment:
This part is a little bit tricky, as it exercises a whole set of code that is not necessarily required and also potentially problematic (worth probably to be reviewed into another PR @bfournie).
Essentially, it re-retriggers the download of the coreos image from the internet, even thought the `BaseIso` dependency already extracted it from the release payload:

```
level=debug msg=  Generating BaseIso Image...
level=info msg=Extracting base ISO from release payload
...
level=debug msg=Extracted base ISO image ~/.cache/agent/image_cache/coreos-x86_64.iso from release payload
```
and immediately after:
```
level=debug msg=Generating Agent Installer PXE Files...
level=debug msg=Obtaining RHCOS image file from 'https://rhcos.mirror.openshift.com/art/storage/prod/streams/4.13/builds/413.86.202212131234-0/x86_64/rhcos-413.86.202212131234-0-live.x86_64.iso'
level=debug msg=Unpacking file into "~/.cache/agent/image_cache/rhcos-413.86.202212131234-0-live.x86_64.iso"...
```

By debugging the code, that seems to happen because the image gets initially saved as `coreos-x86_64.iso` in the cache, thus the `GetIsoPluggable` re-downloads it again since it was looking for a `rhcos-413.86.202212131234-0-live.x86_64.iso` in the cache.

Apart this issue (and as I said I think we should review that portion of the code in the BaseIso), in the current asset the only piece of information required is the location of the base ISO. I think it would be safer than to expose this info directly as a member of the `BaseIso` and use it here

Reply:
Done